### PR TITLE
Take out test cases

### PR DESCRIPTION
**Pull request by andrey-p**
**Reviewer:** @jdorweiler 
**CC:** @MariagraziaAlastra @dharb @brindy 

## Description:
Takes out the test cases for the grade calculation.

I also had the choice between updating the old tests, or taking advantage of the fact that the grade class is really unit testable, and finally putting together a test framework that we can run using the command line.

I went for the latter.

## Steps to test this PR:

1. `npm install`
2. `npm run test`

## Automated tests:
- [x] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: unit-test/data/grade-cases.json [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+      "values": {"hasHTTPS":true, "isaMajorTrackingNetwork": true, "inMajorTrackingNetwork":true, "totalBlocked": 1, "hasObscureTracker": false, "tosdr": {"class": "C"}},
+      "result": {"before": "D", "after": "C", "beforeIndex": 4, "afterIndex": 2 },
+      "descr": "is major tracking network, 1 tracker blocked, tosdr class C"
+  },
+  {
+      "values": {"hasHTTPS":true, "isaMajorTrackingNetwork": true, "inMajorTrackingNetwork":false, "totalBlocked": 0, "hasObscureTracker": false, "tosdr": {"class": "D"}},
+      "result": {"before": "D", "after": "D", "beforeIndex": 3, "afterIndex": 3 },
+      "descr": "is major tracking network, tosdr class D"
+  }
+]
```
[the remark this answers is not in the document]
Comment (brindy) [reply unit-test/data/grade-cases.json]: That's great thanks. Russell mentioned getting the raw score in as well, will that come later?

[the remark this answers is not in the document]
Comment (andrey-p) [reply unit-test/data/grade-cases.json]: It's there - `beforeIndex` and `afterIndex`. I went with what they were called in the extension code, but maybe `beforeScore` and `afterScore` would be better names.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+language: node_js
+node_js:
+  - 0.8
```
Comment (andrey-p) [inline .travis.yml]: I think this'll start going once this file hits `master`. https://travis-ci.org/duckduckgo/duckduckgo-privacy-extension

**Comment by jdorweiler** [comment]
👍

**Pull request merged**